### PR TITLE
Format precommit hook

### LINKED ISSUE #2486: Automatic Formatter Pre-Commit Hook?
Related to #2292, it would be great if there was some way to automatically detect (and fix?) some of the most egregious violations of the [conventions](https://usaco.guide/general/adding-solution?lang=cpp#general) (at the very least, spaces vs tabs).

### PR DESCRIPTION
**Pull request by bqi343**
Formats everything, closes #2486.

_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/bronze/usaco-761.mdx [hunk 1]
```diff
@@ -162,10 +158,10 @@ public class Measurement {
 <PySection>
 
 ```py
-NAMES = ['Bessie', 'Elsie', 'Mildred']
+NAMES = ["Bessie", "Elsie", "Mildred"]
```
Comment (SansPapyrus683) [inline solutions/bronze/usaco-761.mdx:161]: so we're using double quotes for python now, huh?

Comment (bqi343) [reply solutions/bronze/usaco-761.mdx:161]: yes

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    -   id: check-yaml
+    -   id: check-added-large-files
+- repo: local
+  hooks:
+    - id: format-mdx
+      name: format-mdx
+      entry: python py/format_code.py
+      language: python
+      types: [mdx]
+      additional_dependencies: [black-with-tabs, clang-format]
```
Comment (bqi343) [inline .pre-commit-config.yaml:16]: this works for me but idk if it actually installs clang-format correctly

File: solutions/bronze/usaco-761.mdx (hunk 1 above)
Comment (bqi343) [reply solutions/bronze/usaco-761.mdx:161]: yes

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+repos:
```
Comment (bqi343) [inline .pre-commit-config.yaml:1]: @thecodingwizard this look okay to you?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (bqi343) [reply .pre-commit-config.yaml:16]: someone should check whether this works in on their repo

File: .pre-commit-config.yaml (hunk 3 above)
Comment (thecodingwizard) [reply .pre-commit-config.yaml:1]: sure LOL (i don't really know what they do)

**Review by thecodingwizard — APPROVED** [review]
this seems fine (I didn't look too closely though). should probably merge faster to avoid merge conflicts

**Pull request merged**